config/config.go

```diff
@@ -25,195 +25,230 @@ import (
+	// Scheme represents the HTTP URL scheme of the proxy destination, default is http.
 	Scheme string `yaml:"scheme"`
 
-	// Host represent the proxy destination host, for example localhost.
+	// Host represents the proxy destination host, for example, localhost.
 	Host string `yaml:"host"`
 
-	// Port represent the proxy destination port number.
+	// Port represents the proxy destination port number.
 	Port uint16 `yaml:"port"`
 
-	// RoleHeader represent the HTTP header key name of the role token for Role token proxy request.
-	RoleHeader string `yaml:"role_header_key"`
-
-	// BufferSize represent the reverse proxy buffer size.
-	BufferSize uint64 `yaml:"buffer_size"`
+	// BufferSize represents the reverse proxy buffer size.
+	BufferSize uint64 `yaml:"bufferSize"`
 
-	// BypassURLPaths represent URL paths that require authorization bypassing.
-	// WARNING!!! Setting this configuration may introduce security hole in your system. Do NOT set this configuration with URL paths other than application's healthcheck endpoint.
-	// Tips for performance: define your healthcheck endpoint with different lenght from the most frequestly used endpoint, e.g. `/most_used` (len: 10) with `/healthcheck` (len: 12), instead of `/healthccc` (len: 10)
-	BypassURLPaths []string `yaml:"bypass_url_paths"`
+	// BypassURLPaths represents URL paths that require to bypass authorization.
+	// WARNING!!! Setting this configuration may introduce security hole in your system. ONLY set this configuration as the application's health check endpoint.
+	// Tips for performance: define your health check endpoint with a different length from the most frequestly used endpoint, for example, use `/healthcheck` (len: 12) when `/most_used` (len: 10), instead of `/healthccc` (len: 10)
+	BypassURLPaths []string `yaml:"bypassUrlPaths"`
 }
 
-// Authorization represents the detail configuration of the authorization proxy.
+// Authorization represents the detail authorization configuration.
 type Authorization struct {
-	// PubKeyRefreshDuration represents the refresh duration of Athenz PubKey.
-	PubKeyRefreshDuration string `yaml:"pubKeyRefreshDuration"`
+	// AthenzDomains represents Athenz domains containing the RBAC policies.
+	AthenzDomains []string `yaml:"athenzDomains"`
 
-	// PubKeySysAuthDomain represents the system authentication domain of Athenz.
-	PubKeySysAuthDomain string `yaml:"pubKeySysAuthDomain"`
+	// PublicKey represents the configuration to fetch Athenz public keys.
+	PublicKey PublicKey `yaml:"publicKey"`
 
-	// PubKeyEtagExpTime represents the Etag cache expiration time of Athenz PubKey.
-	PubKeyEtagExpTime string `yaml:"pubKeyEtagExpTime"`
+	// Policy represents the configuration to fetch Athenz policies.
+	Policy Policy `yaml:"policy"`
 
-	// PubKeyEtagFlushDur represent the Etag cache expiration check duration.
-	PubKeyEtagFlushDur string `yaml:"pubKeyEtagFlushDur"`
+	// JKWS represents the configuration to fetch Athenz JKWS.
+	JKWS JKWS `yaml:"jwks"`
```

Comment:
Are you intending to `JWK`?
Also, since token is singular, I thought `jwk` would be fine, but why do you use `jwks`? 
```suggestion
	JWKS JWKS `yaml:"jwks"`
```

Reply:
It refers to `JSON Web Key Sets`. https://tools.ietf.org/html/rfc7517#appendix-A
But, seems `jwk` is simpler.

Reply:
> JSON Web Key Sets

You didn't mean plural, thank you.